lte/gateway/python/magma/pipelined/app/classifier.py

```diff
@@ -0,0 +1,253 @@
+
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+
+        # Delete ARP flow for LOCAL port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
+                           in_port=self._uplink_port, arp_tpa=ue_ip_adr)
+
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+
+        # Delete ARP flow for mtr port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
+                           in_port=self.config.mtr_port, arp_tpa=ue_ip_adr)
+
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+
+
+    def _discard_tunnel_flows(self, precedence:int, i_teid:int,
+                              ue_ip_adr:str):
+        priority = self.convert_precedence_to_priority(precedence)
+        # discard uplink Tunnel
+        match = MagmaMatch(tunnel_id=i_teid, in_port=self.config.gtp_port)
+        cookie = self._cookie
+
+        flows.add_flow(self._datapath, self.tbl_num, match,
+                       priority=priority + 1,cookie=cookie)
+
+        # discard downlink Tunnel for LOCAL port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           in_port=self._uplink_port, ipv4_dst=ue_ip_adr)
+        cookie = self._cookie + 1
+
+        flows.add_flow(self._datapath, self.tbl_num, match,
+                       priority=priority + 1,cookie=cookie)
+
+        # discard downlink Tunnel for mtr port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           in_port=self.config.mtr_port,ipv4_dst=ue_ip_adr)
+        cookie = self._cookie + 1
+ 
+        flows.add_flow(self._datapath, self.tbl_num, match,
+                       priority=priority + 1,cookie=cookie)
+
+    def _resume_tunnel_flows(self, precedence:int, i_teid:int,
+                             ue_ip_adr:str):
+
+        priority = self.convert_precedence_to_priority(precedence)
+        # Forward flow for gtp port
+        match = MagmaMatch(tunnel_id=i_teid, in_port=self.config.gtp_port)
+        cookie = self._cookie
+        cookie_mask = self._cookie
+
+        flows.delete_flow(self._datapath, self.tbl_num, match,
+                          priority=priority + 1,
+                          cookie=cookie, cookie_mask=cookie_mask)
+
+        # Forward flow for LOCAL port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           in_port=self._uplink_port,ipv4_dst=ue_ip_adr)
+        cookie = self._cookie + 1
+        cookie_mask = self._cookie + 1
+
+        flows.delete_flow(self._datapath, self.tbl_num, match,
+                          priority=priority +1,
+                          cookie=cookie, cookie_mask=cookie_mask)
+
+        # Forward flow for mtr port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           in_port=self.config.mtr_port,ipv4_dst=ue_ip_adr)
+        cookie = self._cookie + 1
+        cookie_mask = self._cookie + 1
```

Comment:
why do we need cookie?

Reply:
As OAI code, for "discard_downlink_tunnel_flow", the cookie is used for matching criteria as below:
 // match all ports and groups
  downlink_fm.out_port(of13::OFPP_ANY);
  downlink_fm.out_group(of13::OFPG_ANY);
  downlink_fm.cookie(cookie + 1);
  downlink_fm.cookie_mask(cookie + 1);

So that we are using for same.

Reply:
For convert_precedence_to_priority remark, the function "PolicyMixin._get_of_priority()" is specific to convert for enforcement priority like:
# Effectively range is 3 -> 65535
 ENFORCE_PRIORITY_RANGE = MAX_ENFORCE_PRIORITY - MIN_ENFORCE_PROGRAMMED_FLOW

Reply:
I dont think we need to follow every flow parameter as OAI code, we need to check if specific parameter is used and make decision.

Reply:
Yes. We did not get the test cases for CSFB functionality check. So could not get the use case of cookie parameter.
Now we will remove the cookie parameters for MVC scope. Later we will do investigate.